src/openvic-simulation/pop/Pop.cpp

```diff
@@ -162,6 +172,59 @@ void Pop::update_gamestate(
 //TODO store income
 void Pop::add_rgo_owner_income(const fixed_point_t income) {}
 void Pop::add_rgo_worker_income(const fixed_point_t income) {}
+//adds income to the temporary stores and then to general income store
+void Pop::add_artisanal_income(const fixed_point_t income){
+	artisanal_income += income;
+	this->income += income;
+}
+void Pop::add_factory_worker_income(const fixed_point_t income){
+	this->factory_worker_income += income;
+	this->income += income;
+}
+void Pop::add_factory_owner_income(const fixed_point_t income){
+	this->factory_owner_income += income;
+	this->income += income;
+}
+void Pop::add_unemployment_subsidies_income(const fixed_point_t income){
+	this->unemployment_subsidies_income += income;
+	this->income += income;
+}
+void Pop::add_pensions_income(const fixed_point_t income){
+	this->pensions_income += income;
+	this->income += income;
+}
+void Pop::add_government_salary_administration_income(const fixed_point_t income){
+	this->government_salary_administration_income += income;
+	this->income += income;
+}
+void Pop::add_government_salary_education_income(const fixed_point_t income){
+	this->government_salary_education_income += income;
+	this->income += income;
+}
+void Pop::add_government_salary_military_income(const fixed_point_t income){
```

Comment:
These are a lot of very similar methods. If you want, you can refactor them using a macro.
A macro is basically build in code generation.

Reply:
Very good idea, could also be used in the variable and function declarations to streamline the whole process. We can discuss this tomorrow if you want any guidance on it 🙂